src/app-logic/__mocks__/l10n.js

```diff
@@ -0,0 +1,21 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+
+/* global jest */
+const realL10n = jest.requireActual('firefox-profiler/app-logic/l10n.js');
+
+// This mocks en-US only.
+async function mockFetchMessages(locale = 'en-US') {
+  // Always return the english locale, whatever the requested locale
+  const fs = require('fs').promises;
+  const messages = await fs.readFile('../../../locales/en-US/app.ftl', 'UTF-8');
+  return [locale, messages];
+}
+
+module.exports = {
+  ...realL10n,
+  fetchMessages: jest.fn(mockFetchMessages),
+};
```

Comment:
This works for me:
```diff
diff --git a/src/app-logic/__mocks__/l10n.js b/src/app-logic/__mocks__/l10n.js
index d6359d94..d886bed7 100644
--- a/src/app-logic/__mocks__/l10n.js
+++ b/src/app-logic/__mocks__/l10n.js
@@ -6,16 +6,18 @@
 
 /* global jest */
 const realL10n = jest.requireActual('firefox-profiler/app-logic/l10n.js');
 
 // This mocks en-US only.
 async function mockFetchMessages(locale = 'en-US') {
   // Always return the english locale, whatever the requested locale
   const fs = require('fs').promises;
-  const messages = await fs.readFile('../../../locales/en-US/app.ftl', 'UTF-8');
+  // Note: with yarn, the current directory is always the root of the project,
+  // so we can use a relative path.
+  const messages = await fs.readFile('./locales/en-US/app.ftl', 'UTF-8');
   return [locale, messages];
 }
 
 module.exports = {
   ...realL10n,
-  fetchMessages: jest.fn(mockFetchMessages),
+  fetchMessages: mockFetchMessages,
 };
```

There was 2 problems:
1. `jest.fn` was doing wrong things, but I don't understand why really... The function wasn't being used.
2. The path was incorrect, because the current directory isn't this file's directory, but the root directory of the project. I changed that.

Now with this, I can comment the line with `unmock` and the test for AppLocalisationProvider fails with the fallback string. Then you can move forward and use one of the ID in the en-US FTL file and look if this is properly localized.